src/screens/SendConfirmationSheet.tsx

```diff
@@ -242,28 +261,132 @@ export default function SendConfirmationSheet() {
     return get(contacts, `${[toLower(toAddress)]}`);
   }, [contacts, toAddress]);
 
-  const [checkboxes, setCheckboxes] = useState([
-    {
-      checked: false,
-      label: lang.t(
-        'wallet.transaction.checkboxes.im_not_sending_to_an_exchange'
-      ),
-    },
-    {
-      checked: false,
-      label: lang.t(
-        'wallet.transaction.checkboxes.has_a_wallet_that_supports',
-        {
-          networkName: capitalize(network),
+  const uniqueTokenType = getUniqueTokenType(asset);
+  const isENS = uniqueTokenType === 'ENS';
+
+  const [checkboxes, setCheckboxes] = useState<Checkbox[]>(
+    isENS
+      ? ([
+          !hasClearProfileInfo(ensProfile) && {
+            checked: false,
+            id: 'clear-records',
+            label: 'Clear profile information',
+          },
+          !doesNamePointToRecipient(ensProfile, toAddress) && {
+            checked: false,
+            id: 'set-address',
+            label: 'Point this name to the recipient’s wallet address',
+          },
+          doesAccountControlName(ensProfile) && {
+            checked: false,
+            id: 'transfer-control',
+            label: 'Transfer control to the recipient',
+          },
+        ].filter(Boolean) as Checkbox[])
+      : [
+          {
+            checked: false,
+            id: 'not-sending-to-exchange',
+            label: lang.t(
+              'wallet.transaction.checkboxes.im_not_sending_to_an_exchange'
+            ),
+          },
+          {
+            checked: false,
+            id: 'has-wallet-that-supports',
+            label: lang.t(
+              'wallet.transaction.checkboxes.has_a_wallet_that_supports',
+              {
+                networkName: capitalize(network),
+              }
+            ),
+          },
+        ]
+  );
+
+  useEffect(() => {
+    if (isENS) {
+      const promises = [
+        estimateGasLimit(
+          {
+            address: accountAddress,
+            amount: 0,
+            asset: asset,
+            recipient: toAddress,
+          },
+          true
+        ),
+      ];
+      const sendENSOptions = Object.fromEntries(
+        checkboxes.map(option => [option.id, option.checked])
+      );
+      const cleanENSName = asset?.name?.split(' ')?.[0] ?? asset?.name;
+
+      if (sendENSOptions['clear-records']) {
```

Comment:
wondering if we should put these keys in const outside of the component, i see it being used 4 times

Reply:
Not sure if we need constants in TypeScript? `sendENSOptions` is guards it's keys, so if something else is entered, it will throw a type error